analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.IsNull.cs

```diff
@@ -62,9 +62,9 @@ public void IsNull_Coalesce_UnknownToNull()
 Tag(""UnknownToNull"", unknownToNull);";
             var validator = SETestContext.CreateCS(code, ", object arg").Validator;
             validator.ValidateContainsOperation(OperationKind.IsNull);
-            validator.TagValues("UnknownToNull").Should().HaveCount(2)
-                .And.ContainSingle(x => x == null)
-                .And.ContainSingle(x => x != null && x.HasConstraint(ObjectConstraint.Null));
+            validator.TagValues("UnknownToNull").Should().SatisfyRespectively(
+                x => x.Should().HaveOnlyConstraint(ObjectConstraint.Null),      // arg was null, and got coalesce value of nullValue
+                x => x.Should().HaveOnlyConstraint(ObjectConstraint.NotNull));  // arg was not null, no coalescing
         }
```

Comment:
I would prefer if you would `Tag("AfterColaescing")` and assert the Constraints of all three symbols for both branches instead. This would also make the comments obsolete.